File: pdl/infra/java/src/main/java/io/pdl/infrastructure/data/DBStore.java
package io.pdl.infrastructure.data;

import java.util.List;
import java.util.Map;

public interface DBStore {
    Map<String, Object> insert(String table, String primaryKey, Map<String, Object> values) throws Exception;

    void update(String table, String primaryKey, Map<String, Object> values) throws Exception;

    void delete(String table, String primaryKey, Map<String, Object> values) throws Exception;

    List<Map<String, Object>> select(
            String table,
            List<Filter> filters,
            List<String> projection,
            List<OrderClause> orderings,
            Integer limit,
            Integer offset) throws Exception;
}

File: pdl/infra/php/src/Io/Pdl/Infrastructure/Data/DBStore.php
<?php

namespace Io\Pdl\Infrastructure\Data;

interface DBStore
{
    public function insert(string $table, string $primaryKey, array $values): array;

    public function update(string $table, string $primaryKey, array $values): void;

    public function delete(string $table, string $primaryKey, array $values): void;

    public function select(
        string $table,
        array $filters,
        array $projection,
        array $orderings,
        ?int $limit,
        ?int $offset
    ): array;
}
